src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Helper/Base64DecoderHelper.cs

```diff
@@ -0,0 +1,1446 @@
+                {
+                    // For overflow see comment below
+                    maxSrcLength = destLength / 3 * 4;
+                }
+
+                T* src = srcBytes;
+                byte* dest = destBytes;
+                T* srcEnd = srcBytes + (uint)srcLength;
+                T* srcMax = srcBytes + (uint)maxSrcLength;
+
+#if NETCOREAPP
+                if (maxSrcLength >= 24)
+                {
+                    T* end = srcMax - 88;
+                    if (Vector512.IsHardwareAccelerated && Avx512Vbmi.IsSupported && (end >= src))
+                    {
+                        Avx512Decode(decoder, ref src, ref dest, end, maxSrcLength, destLength, srcBytes, destBytes);
+
+                        if (src == srcEnd)
+                        {
+                            goto DoneExit;
+                        }
+                    }
+
+                    end = srcMax - 45;
+                    if (Avx2.IsSupported && (end >= src))
+                    {
+                        Avx2Decode(decoder, ref src, ref dest, end, maxSrcLength, destLength, srcBytes, destBytes);
+
+                        if (src == srcEnd)
+                        {
+                            goto DoneExit;
+                        }
+                    }
+
+                    end = srcMax - 66;
+                    if (AdvSimd.Arm64.IsSupported && (end >= src))
+                    {
+                        AdvSimdDecode(decoder, ref src, ref dest, end, maxSrcLength, destLength, srcBytes, destBytes);
+
+                        if (src == srcEnd)
+                        {
+                            goto DoneExit;
+                        }
+                    }
+
+                    end = srcMax - 24;
+                    if ((Ssse3.IsSupported || AdvSimd.Arm64.IsSupported) && BitConverter.IsLittleEndian && (end >= src))
+                    {
+                        Vector128Decode(decoder, ref src, ref dest, end, maxSrcLength, destLength, srcBytes, destBytes);
+
+                        if (src == srcEnd)
+                        {
+                            goto DoneExit;
+                        }
+                    }
+                }
+#endif
+
+                // Last bytes could have padding characters, so process them separately and treat them as valid only if isFinalBlock is true
+                // if isFinalBlock is false, padding characters are considered invalid
+                int skipLastChunk = isFinalBlock ? 4 : 0;
+
+                if (destLength >= decodedLength)
+                {
+                    maxSrcLength = srcLength - skipLastChunk;
+                }
+                else
+                {
+                    // This should never overflow since destLength here is less than int.MaxValue / 4 * 3 (i.e. 1610612733)
+                    // Therefore, (destLength / 3) * 4 will always be less than 2147483641
+                    Debug.Assert(destLength < (int.MaxValue / 4 * 3));
+#if NETCOREAPP
```

Comment:
With the exception of the code in these ifdefs, and with calls being prefixed with `decoder.`, this whole file is an exact copy of what was in the other file, right? Or are there other changes?

Reply:
Exactly as you said,, only changes are if-defs for netsandard, and type parameter to `decoder` no new changes. Well, there is small bugfix that related to the new tests added, nothing else new

Basically all internal members from `Base64Decoder` and `Base64Encoder` moved in this `Base64Herlper` type to avoid conflict referencing `Base64` from netstandard project between .NET 9 and netstandard versions.

Reply:
> Well, there is small bugfix that related to the new tests added, nothing else new

What is that fix?

Reply:
I was fixed the bug found with Base64Url fuzzing only for byte overloads, the same bug left for char overloads.
 
The diff is here https://github.com/dotnet/runtime/pull/103617/commits/73acb51861f24f32ec1f2b3d2007486ca3ebbd42